server/storage/storage_impl.go

```diff
@@ -101,33 +148,151 @@ func (s *MetaStorageImpl) ListSchemas(ctx context.Context, clusterID uint32) ([]
 }
 
 func (s *MetaStorageImpl) PutSchemas(ctx context.Context, clusterID uint32, schemas []*metapb.Schema) error {
+	for _, item := range schemas {
+		key := makeSchemaKey(clusterID, item.Id)
+		value, err := proto.Marshal(item)
+		if err != nil {
+			return ErrMetaPutSchemas.WithCausef("proto parse err: %v", err)
+		}
+		if err = s.Put(ctx, key, string(value)); err != nil {
+			return errors.Wrapf(err, "put schemas err, key: %v", key)
+		}
+	}
 	return nil
 }
 
 func (s *MetaStorageImpl) ListTables(ctx context.Context, clusterID uint32, schemaID uint32, tableID []uint64) ([]*metapb.Table, error) {
-	return nil, nil
+	tables := make([]*metapb.Table, 0)
+	for _, item := range tableID {
+		key := makeTableKey(clusterID, schemaID, item)
+		value, err := s.Get(ctx, key)
+		if err != nil {
+			return nil, errors.Wrapf(err, "get tables err, key: %v", key)
+		}
+		tableData := &metapb.Table{}
+		if err = proto.Unmarshal([]byte(value), tableData); err != nil {
+			return nil, ErrMetaGetTables.WithCausef("proto parse err: %v", err)
+		}
+		tables = append(tables, tableData)
+	}
+	return tables, nil
 }
 
 func (s *MetaStorageImpl) PutTables(ctx context.Context, clusterID uint32, schemaID uint32, tables []*metapb.Table) error {
+	for _, item := range tables {
```

Comment:
Putting all the tables in a batch may be more elegant and effective.

Reply:
Todo, relating to #24

Reply:
If the batch put is not planned to be done in this PR, maybe we should add a TODO for that.